tests/src/core/testqgscoordinatereferencesystem.cpp

```diff
@@ -600,18 +599,39 @@ void TestQgsCoordinateReferenceSystem::equalityInvalid()
   QgsCoordinateReferenceSystem invalidCrs2;
   QVERIFY( invalidCrs1 == invalidCrs2 );
 }
-void TestQgsCoordinateReferenceSystem::readXml()
+void TestQgsCoordinateReferenceSystem::readWriteXml()
 {
-  //QgsCoordinateReferenceSystem myCrs;
-  //myCrs.createFromSrid( GEOSRID );
-  //QgsCoordinateReferenceSystem myCrs2;
-  //QVERIFY( myCrs2.readXml( QDomNode & node ) );
-}
-void TestQgsCoordinateReferenceSystem::writeXml()
-{
-  //QgsCoordinateReferenceSystem myCrs;
-  //bool writeXml( QDomNode & node, QDomDocument & doc ) const;
-  //QVERIFY( myCrs.isValid() );
+  QgsCoordinateReferenceSystem myCrs;
+  myCrs.createFromSrid( GEOSRID );
+  QVERIFY( myCrs.isValid() );
+  QDomDocument document( "test" );
+  QDomElement node = document.createElement( QStringLiteral( "crs" ) );
+  document.appendChild( node );
+  QVERIFY( myCrs.writeXml( node, document ) );
+  QgsCoordinateReferenceSystem myCrs2;
+  QVERIFY( myCrs2.readXml( node ) );
+  QVERIFY( myCrs == myCrs2 );
+
+  // Empty XML made from writeXml operation
+  QgsCoordinateReferenceSystem myCrs3;
+  QDomDocument document2( "test" );
+  QDomElement node2 = document2.createElement( QStringLiteral( "crs" ) );
+  document2.appendChild( node2 );
+  QVERIFY( ! myCrs3.isValid() );
+  QVERIFY( myCrs3.writeXml( node2, document2 ) );
+  QgsCoordinateReferenceSystem myCrs4;
+  QVERIFY( myCrs4.readXml( node2 ) );
+  // These two tests should be enabled, but myCrs4 is hardcoded to be valid while it's not valid!
+//  QVERIFY( ! myCrs4.isValid() );
+//  QVERIFY( myCrs3 == myCrs4 );
```

Comment:
@m-kuhn I'm talking about these disabled tests I'm adding

Reply:
I don't know then... Maybe run the test in a debugger and see what happens?